pkg/operator/ceph/operator.go

```diff
@@ -119,6 +119,10 @@ func (o *Operator) Run() error {
 		if err := rookDiscover.Start(o.operatorNamespace, o.rookImage, o.securityAccount, true); err != nil {
 			return errors.Wrap(err, "failed to start device discovery daemonset")
 		}
+	} else {
```

Comment:
The change looks good, just a small suggestion to follow the same pattern as starting the discovery:
```golang
rookDiscover := discover.New(o.context.Clientset)
if EnableDiscoveryDaemon {
  rookDiscover.Start(...)
} else {
  rookDiscover.Stop(...)
}
```

Reply:
If you move L129 `stopContext, stopFunc := context.WithCancel(context.Background())` up you can use `stopContext` and pass it to `discover.Stop()` that would help for the future. So you don't need to create the `context.TODO()`.

Reply:
Both fixed.